### ISSUE TITLE
Slack App and OAuth Redirect URL

### ISSUE THREAD

**Issue by Vad1mo** [rampatra/jbot#110]
I have two workspaces where I want to use my jbot app, this means I have to distribute my slack app across multiple workspaces. eg. https://api.slack.com/docs/slack-button

There is one mandatory requirement, in order to activate public distribution. There should be a Redirect URL.

> Redirect URLs
You will need to configure redirect URLs in order to automatically generate the Add to Slack button or to distribute your app. If you pass a URL in an OAuth request, it must (partially) match one of the URLs you enter here. Learn more

I didn't see any support for that in jbot nor did I find a redirect jbot path.

Can jobt work across multiple workspaces or do I miss something?

**Comment by rampatra** [comment]
No, you don't need to release your bot as a slack app for the public. If you're releasing it to the public then, of course, you need to specify the redirect urls. But in your case, if you only want to install the bot to two workspaces, you don't need to specify any redirect urls.

Just create a slack app, add a bot integration and then install in the respective workspace. After these steps are done, you can get your token from the "Oauth & Permissions" tab. That's it. I personally have installed jbot across multiple workspaces without the need of any redirect urls. All you need is the Bot token.

**Comment by Vad1mo** [comment]
thx,
So you are deploying a new jbot instance for each workspace. As you create an app in each workspace you have also two bot keys. Am I right?

**Comment by rampatra** [comment]
Yes, you're right. I have two keys and have two instances of JBot (separate bots).

**Comment by rampatra** [comment]
If you want to use the same bot across workspaces then you have to make a slack app first and then add a bot integration to it. This is enough to install your app (or bot) to install in workspaces which you have rights to. But to make it public, I think you need to add more configurations. I will give it a look and update this ticket.
